### PR TITLE
Feat/st 662: Record Listing - Hearing Venues

### PR DESCRIPTION
**Pull request by sjamihmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ST-662

### Change description ###
Changes for Hearing venues in record listing flow

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by muradalamv1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java [hunk 1]
```diff
@@ -34,17 +47,37 @@ public void configure(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
...
         addOtherInformation(pageBuilder);
     }
 
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(CaseDetails<CaseData, State> details) {
+        var caseData = details.getData();
+        DynamicList regionList = locationService.getAllRegions();
+        caseData.getRecordListing().setRegionList(regionList);
+
+        if (regionList.getListItems().isEmpty()) {
+            caseData.getRecordListing().setRegionsMessage("Unable to retrieve Region data");
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java:70]: Style thing, but could use ternary operator here?

`if (regionList.getListItems().isEmpty()) ? caseData.getRecordListing().setRegionsMessage("Unable to retrieve Region data") : caseData.getRecordListing().setRegionsMessage("");`

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java [hunk 2]
```diff
@@ -65,15 +98,52 @@ public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
...
+                                                                  CaseDetails<CaseData, State> detailsBefore) {
+        final CaseData caseData = details.getData();
+        String selectedRegion = caseData.getRecordListing().getSelectedRegionVal();
+        String regionId = getRegionId(selectedRegion);
+
+        if (null != regionId) {
+            DynamicList hearingVenueList = locationService.getHearingVenuesByRegion(regionId);
+            caseData.getRecordListing().setHearingVenues(hearingVenueList);
+            if (hearingVenueList.getListItems().isEmpty()) {
+                caseData.getRecordListing().setHearingVenuesMessage("Unable to retrieve Hearing Venues data");
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java:111]: As above, maybe use ternary operator?

File: src/test/java/uk/gov/hmcts/sptribs/recordlisting/LocationServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    @Mock
+    private HttpServletRequest httpServletRequest;
+
+    @Mock
+    private LocationClient locationClient;
+
+    @Test
+    void shouldPopulateRegionDynamicList() {
+        //When
+        when(authTokenGenerator.generate()).thenReturn(TEST_SERVICE_AUTH_TOKEN);
```
Comment (osheaf01) [inline src/test/java/uk/gov/hmcts/sptribs/recordlisting/LocationServiceTest.java:46]: Can you make the comments //Given //When //Then in this class, as you've done above in CaseworkerRecordListingTest please? Makes it more readable if we use a consistent standard.

File: src/test/java/uk/gov/hmcts/sptribs/recordlisting/LocationServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+        DynamicList regionList = locationService.getAllRegions();
+
+        assertThat(regionList.getListItems()).isEmpty();
+    }
+
+    @Test
+    void shouldPopulateHearingVenueDynamicList() {
+        //When
+        when(authTokenGenerator.generate()).thenReturn(TEST_SERVICE_AUTH_TOKEN);
```
Comment (osheaf01) [inline src/test/java/uk/gov/hmcts/sptribs/recordlisting/LocationServiceTest.java:81]: //Given //When //Then

File: src/test/java/uk/gov/hmcts/sptribs/recordlisting/LocationServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        //Then
+        assertThat(hearingVenueList).isNotNull();
+        assertThat(hearingVenueList.getListItems()).extracting("label").containsExactlyInAnyOrder("courtName-courtAddress");
+
+    }
+
+    @Test
+    void shouldPopulateHearingVenueDynamicListWithMissingCourtAddress() {
+        //When
+        when(authTokenGenerator.generate()).thenReturn(TEST_SERVICE_AUTH_TOKEN);
```
Comment (osheaf01) [inline src/test/java/uk/gov/hmcts/sptribs/recordlisting/LocationServiceTest.java:105]: //Given //When //Then

**Review by osheaf01 — CHANGES_REQUESTED** [review]
Mostly fine, please amend the test comments. Ternary operator comments optional.

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java (hunk 1 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java]: Done. Used ternary operator now,

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java (hunk 2 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java]: Done. Used ternary operator now,

File: src/test/java/uk/gov/hmcts/sptribs/recordlisting/LocationServiceTest.java (hunk 3 above)
Comment (sjamihmcts) [reply src/test/java/uk/gov/hmcts/sptribs/recordlisting/LocationServiceTest.java]: Done.

**Review by osheaf01 — APPROVED** [review]
LGTM

**Review by muradalamv1 — APPROVED** [review]

**Pull request merged**